dash-pipeline/SAI/templates/saifixedapis.cpp.j2

```diff
@@ -2,136 +2,130 @@
-        
-        case SAI_SWITCH_ATTR_DEFAULT_1Q_BRIDGE_ID:
-            attr->value.oid = DASH_BMV2_DEFAULT_1Q_BRIDGE_ID;
-            fprintf(stderr, "  sai_get_switch_attribute() [0] attr %d SAI_SWITCH_ATTR_DEFAULT_1Q_BRIDGE_ID = %lx\n", attr->id, attr->value.oid);
-            return SAI_STATUS_SUCCESS;
-
-        case SAI_SWITCH_ATTR_CPU_PORT:
-            attr->value.oid = DASH_BMV2_DEFAULT_CPU_PORT_ID;
-            fprintf(stderr, "  sai_get_switch_attribute() [%d] attr %d DASH_BMV2_DEFAULT_CPU_PORT_ID = %lx\n",
-                    i, attr->id, attr->value.oid);
-            return SAI_STATUS_SUCCESS;
 
-        default:
-            fprintf(stderr, "  sai_get_switch_attribute() [0] attr %d is NOT SUPPORTED - returning SAI_STATUS_SUCCESS anyway\n", attr->id);
-            return SAI_STATUS_SUCCESS;
+    for (uint32_t i = 0; i < attr_count ; i++, attr++)
+    {
+        switch(attr->id)
+        {
+            case SAI_SWITCH_ATTR_NUMBER_OF_ACTIVE_PORTS:
+
+                attr->value.u32 = DASH_BMV2_NUM_PORTS;
+
+                DASH_LOG_NOTICE("[%d] attr %d SAI_SWITCH_ATTR_NUMBER_OF_ACTIVE_PORTS = %d", i, attr->id, attr->value.u32);
+
+                break;
+
+            case SAI_SWITCH_ATTR_PORT_LIST:
+
+                attr->value.objlist.count = DASH_BMV2_NUM_PORTS;
+
+                for (uint32_t j = 0; j < DASH_BMV2_NUM_PORTS; j++)
+                {
+                    attr->value.objlist.list[j] = port_list[j];
+                }
+
+                DASH_LOG_NOTICE("[%d] attr %d SAI_SWITCH_ATTR_PORT_LIST = [%d objids]", i, attr->id, DASH_BMV2_NUM_PORTS);
+
+                break;
+
+            case SAI_SWITCH_ATTR_DEFAULT_VLAN_ID:
+
+                attr->value.oid = DASH_BMV2_DEFAULT_VLAN_ID;
+
+                DASH_LOG_NOTICE("[%d] attr %d SAI_SWITCH_ATTR_DEFAULT_VLAN_ID = %lx", i, attr->id, attr->value.oid);
+
+                break;
+
+            case SAI_SWITCH_ATTR_DEFAULT_VIRTUAL_ROUTER_ID:
+
+                attr->value.oid = DASH_BMV2_DEFAULT_VRF_ID;
+
+                DASH_LOG_NOTICE("attr %d SAI_SWITCH_ATTR_DEFAULT_VIRTUAL_ROUTER_ID = %lx", attr->id, attr->value.oid);
+
+                break;
+
+            case SAI_SWITCH_ATTR_DEFAULT_1Q_BRIDGE_ID:
+
+                attr->value.oid = DASH_BMV2_DEFAULT_1Q_BRIDGE_ID;
+
+                DASH_LOG_NOTICE("attr %d SAI_SWITCH_ATTR_DEFAULT_1Q_BRIDGE_ID = %lx", attr->id, attr->value.oid);
+
+                break;
+
+            case SAI_SWITCH_ATTR_CPU_PORT:
+
+                attr->value.oid = DASH_BMV2_DEFAULT_CPU_PORT_ID;
+
+                DASH_LOG_NOTICE("[%d] attr %d DASH_BMV2_DEFAULT_CPU_PORT_ID = %lx", i, attr->id, attr->value.oid);
+
+                break;
+
+            default:
+
+                DASH_LOG_WARN("attr %d is NOT SUPPORTED, but returning SAI_STATUS_SUCCESS", attr->id);
+
+                memset(&attr->value, 0, sizeof(attr->value)); // clear potential caller garbage
+
+                break; // TODO FIXME should return NOT SUPPORTED
```

Comment:
If I'm not mistaken, all these fixed APIs were added in order to support SAI-PTF discovery functions, even if they aren't really "supported". If you return other that SUCCESS it might break all PTF testing. Try it and see.

Reply:
yes, i already tried that, and those tests fail, but ideally i would like to return here not supported, since in syncd we don't want success on all non existing attributes, we can handle them in different way, but if i return not supported here, then i dont know how to fix ptf tests

Reply:
i think we could add dummy attributes that are used in ptf test cases, or mitigate that on saithrift

Reply:
> add dummy attributes that are used in ptf test cases

That seems reasonable.

Reply:
now how to find all attributes that are used by test ptf test cases? :P